xarray/coding/times.py

```diff
@@ -628,25 +628,38 @@ def _encode_datetime_with_cftime(dates, units: str, calendar: str) -> np.ndarray
     if cftime is None:
         raise ModuleNotFoundError("No module named 'cftime'")
 
+    dates = np.array(dates)
+    original_shape = dates.shape
+
     if np.issubdtype(dates.dtype, np.datetime64):
         # numpy's broken datetime conversion only works for us precision
         dates = dates.astype("M8[us]").astype(datetime)
 
-    def encode_datetime(d):
-        # Since netCDF files do not support storing float128 values, we ensure
-        # that float64 values are used by setting longdouble=False in num2date.
-        # This try except logic can be removed when xarray's minimum version of
-        # cftime is at least 1.6.2.
-        try:
-            return (
-                np.nan
-                if d is None
-                else cftime.date2num(d, units, calendar, longdouble=False)
-            )
-        except TypeError:
-            return np.nan if d is None else cftime.date2num(d, units, calendar)
+    dates = np.atleast_1d(dates)
+
+    # Find all the None position
+    none_position = np.equal(dates, None)
```

Comment:
mypy complains here.
Not sure if this is an issue of wrong typing of numpy, or if this is not a recommended way of calling equals. (I think the None is the problem).